packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -409,6 +412,40 @@ async function removeBotConfig(ctx: CoreHookContext) {
     delete envConfig[ResourcePlugins.Bot];
     envConfig[ResourcePlugins.Bot] = { wayToRegisterBot: "create-new" };
   }
+  let needUpdate = false;
+  let configPrefix = "";
+  if (envConfig["solution"]) {
+    if (envConfig["solution"]["subscriptionId"] && envConfig["solution"]["resourceGroupName"]) {
+      configPrefix = `/subscriptions/${envConfig["solution"]["subscriptionId"]}/resourcegroups/${envConfig["solution"]["resourceGroupName"]}`;
+      needUpdate = true;
+    }
+  }
+  if (needUpdate && envConfig[ResourcePlugins.FrontendHosting]) {
+    envConfig[ResourcePlugins.FrontendHosting][
+      EnvConfigName.StorageResourceId
+    ] = `${configPrefix}/providers/Microsoft.Storage/storageAccounts/${
+      envConfig[ResourcePlugins.FrontendHosting][EnvConfigName.StorageName]
+    }`;
+  }
+  if (needUpdate && envConfig[ResourcePlugins.AzureSQL]) {
```

Comment:
we may need to handle cases that provision partially success: for example, create resource group succeeded but create sql failed because of the quota issue. 
By the way, since 'fx-resource-sql` config (as well as other config) exists in `env.default.json` before we run provision, `envConfig[ResourcePlugins.AzureSQL]` seems will always be true?

Reply:
sure, thanks!